### DISCUSSION TITLE
Questions about the Temporary Library

### DISCUSSION THREAD

**Discussion by zalmang** [codefori/vscode-ibmi#1778] [Q&A] [answered]
1) Why does Code for i insist on a different temporary library that my IBM i user's temporary library (QTEMP)
2) My old temporary library was ILEDITOR. I was able to connect to my system with Code for i, but the object browser wasn't working. I did some digging and found that my user doesn't have access to ILEDITOR. Why was that issue causing specifically the object browser not to work?

**Comment by sebjulliand** [comment, marked as answer]
Hi @zalmang!

1. Because Code for i is stateless. Not by choice, mind you, but because IBM i commands sent through SSH run in a job that gets opened to run them and then ends immediately. So it cannot use QTEMP in this context.
2. In order to list its content, the object browser uses `DSPBJD`/`DSPFD` in combination with `OUTPUT(*OUTFILE)`. The output file is written in the temporary library set at connection's level and then its content is read.
